ingestion-beam/src/main/java/com/mozilla/telemetry/transforms/LimitPayloadSize.java

```diff
@@ -0,0 +1,52 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package com.mozilla.telemetry.transforms;
+
+import org.apache.beam.sdk.io.gcp.pubsub.PubsubMessage;
+import org.apache.beam.sdk.metrics.Counter;
+import org.apache.beam.sdk.metrics.Metrics;
+
+public class LimitPayloadSize extends MapElementsWithErrors.ToPubsubMessageFrom<PubsubMessage> {
+
+  /** Exception to throw for messages whose payloads exceed the configured size limit. */
+  public static class PayloadTooLargeException extends RuntimeException {
```

Comment:
why does this inherit RuntimeException instead of CheckedException?

Reply:
I will change it to inherit from Exception. This is mostly just habit of preferring runtime exceptions since exception checking seems to be largely regarded as a failed feature of Java. Checked is fine in this case.